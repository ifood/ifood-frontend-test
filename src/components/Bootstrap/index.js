import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Media from 'react-bootstrap/Media'
import Modal from 'react-bootstrap/Modal'
import ModalDialog from 'react-bootstrap/ModalDialog'
import ModalHeader from 'react-bootstrap/ModalHeader'
import ModalTitle from 'react-bootstrap/ModalTitle'
import ModalBody from 'react-bootstrap/ModalBody'

export {
  Container,
  Row,
  Col,
  Media,
  Modal,
  ModalDialog,
  ModalHeader,
  ModalTitle,
  ModalBody,
}
