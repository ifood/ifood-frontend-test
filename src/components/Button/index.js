import React from "react";
import PropTypes from "prop-types";

import * as S from "./styled";

const Button = ({ onClick, label, variation }) => {
  return (
    <S.Button variation={variation} onClick={onClick}>
      {label}
    </S.Button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  variation: PropTypes.oneOf(["primary", "secondary"]).isRequired,
};

export default Button;
