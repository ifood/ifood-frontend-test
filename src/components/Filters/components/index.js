/* eslint-disable react/jsx-props-no-spreading */

import React from 'react';
import Amount from './Amount';
import DatePicker from './DatePicker';
import Page from './Page';
import SelectCountry from './SelectCountry';
import SelectLocale from './SelectLocale';

const locale = (props) => <SelectLocale {...props} />;
const timestamp = (props) => <DatePicker {...props} />;
const offset = (props = {}) => <Page {...props} />;
const country = (props) => <SelectCountry {...props} />;
const limit = (props) => <Amount {...props} />;

export default { locale, timestamp, offset, country, limit };
