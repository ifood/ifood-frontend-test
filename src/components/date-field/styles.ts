import styled from 'styled-components';
import colors from 'common/colors';

export const DateField = styled.div`
  .react-datepicker-wrapper {
    width: 100%;
  }

  input {
    height: 26px;
    border: 2px solid ${colors.GRAY_LIGHT};
    border-radius: 4px;
    padding-left: 10px;
    width: calc(99.4% - 10px);
  }
`;
