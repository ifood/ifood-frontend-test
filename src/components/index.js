import Loading from './loading'
import Header from './header'
import Filters from './filters'
import CardFeatured from './card-featured'
import Premium from './premium'

export {
  Loading,
  Header,
  Filters,
  CardFeatured,
  Premium
}
