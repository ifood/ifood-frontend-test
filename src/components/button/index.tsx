import React from 'react';

import * as S from './styles';

export const Button = (): JSX.Element => {
  return <S.Button>Button </S.Button>;
};
