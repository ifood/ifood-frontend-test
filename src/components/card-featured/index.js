import React from 'react'
import { FormattedMessage } from 'react-intl'
import PropTypes from 'prop-types'

import * as S from './styles'

import Background from './background'
import Pagination from './pagination'

export default function CardFeatured ({ title, playlists, searchies, onSubmit }) {
  return (
    <S.Container>
      <S.Content>
        <S.WrapperTitle>
          <S.Title>
            <FormattedMessage id='greeting' values={{ title }}/>
          </S.Title>
          <S.PlaylistTitle>{ title }</S.PlaylistTitle>
        </S.WrapperTitle>
        <Pagination
          totalItems={playlists.total}
          limit={searchies.limit}
          onSubmitPagination={onSubmit}/>
        <S.WrapperCard>
          {playlists.items && playlists.items.map(item => (
            <S.Card key={item.id}>
              <S.LinkSpotify href={item.external_urls.spotify} target='_blank'>
                <Background
                  bg={item.images[0].url}
                  title={item.name}/>
              </S.LinkSpotify>
              <S.TitleItem>{item.name}</S.TitleItem>
            </S.Card>
          ))}
        </S.WrapperCard>
      </S.Content>
    </S.Container>
  )
}

CardFeatured.propTypes = {
  title: PropTypes.string,
  playlists: PropTypes.shape({
    total: PropTypes.number,
    items: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      images: PropTypes.arrayOf(PropTypes.shape({
        url: PropTypes.string
      }))
    }))
  }).isRequired,
  searchies: PropTypes.shape({
    limit: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ])
  }),
  onSubmit: PropTypes.func.isRequired
}
