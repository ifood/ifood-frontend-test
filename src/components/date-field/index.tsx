import React from 'react'; //eslint-disable-line
import DatePicker from 'react-datepicker';

import 'react-datepicker/dist/react-datepicker.css';

import * as S from './styles';

interface IDatePicker {
  startDate: Date;
  onChangeDate(date: Date, event?: React.SyntheticEvent<any>): void;
}

export const DateField: React.FC<IDatePicker> = ({ startDate, onChangeDate }) => {
  return (
    <S.DateField>
      <label htmlFor="date-picker">Date</label>
      <DatePicker selected={startDate} onChange={onChangeDate} dateFormat="dd-MM-yyyy" id="date-picker" />
    </S.DateField>
  );
};
