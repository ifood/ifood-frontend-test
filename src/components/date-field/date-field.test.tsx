import React from 'react';
import { render, screen } from '@testing-library/react';
import { DateField } from 'components/date-field';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

const wrapper = () => render(<DateField startDate={new Date()} onChangeDate={() => {}} />);

test('testing accessibility', async () => {
  const { container } = wrapper();
  const results = await axe(container);

  expect(results).toHaveNoViolations();
});

test('renders component without errors', () => {
  const { container } = wrapper();

  expect(container).toMatchSnapshot();
});
