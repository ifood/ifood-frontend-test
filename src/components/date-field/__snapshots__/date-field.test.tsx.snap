// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component without errors 1`] = `
<div>
  <div
    class="sc-bdnylx iysmyN"
  >
    <label
      for="date-picker"
    >
      Date
    </label>
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          class=""
          id="date-picker"
          type="text"
          value="08-10-2020"
        />
      </div>
    </div>
  </div>
</div>
`;
